analyzer/src/main/kotlin/managers/Npm.kt

```diff
@@ -360,10 +362,11 @@ open class Npm(
         return Pair(id.toCoordinates(), module)
     }
 
-    protected open fun getRemotePackageDetails(workingDir: File, packageName: String): JsonNode {
-        val process = run(workingDir, "view", "--json", packageName)
-        return jsonMapper.readTree(process.stdout)
-    }
+    protected open fun getRemotePackageDetails(workingDir: File, packageName: String): JsonNode =
+        npmViewCache.getOrPut(packageName) {
+            val process = run(workingDir, "view", "--json", packageName)
```

Comment:
Here a potentially long-running process is invoked with the map partially locked. This is probably not ideal.
I will approve this PR because I think it improves the current situation in any case; but there is still room for optimization.